addons/l10n_ro_efactura/models/account_move.py

```diff
@@ -0,0 +1,158 @@
+                    break
+
+    @api.depends('l10n_ro_edi_document_ids', 'l10n_ro_edi_state')
+    def _compute_show_reset_to_draft_button(self):
+        """ Prevent user to reset move to draft when there's an
+            active sending document or an OK response has been received """
+        super()._compute_show_reset_to_draft_button()
+        for move in self:
+            received_error_document = (move.l10n_ro_edi_state == 'invoice_sending' and
+                                       all(doc.state != 'invoice_sending_failed' for doc in move.l10n_ro_edi_document_ids))
+            if move.l10n_ro_edi_state == 'invoice_sent' or received_error_document:
+                move.show_reset_to_draft_button = False
+
+    def _l10n_ro_edi_get_attachment_file_name(self):
+        self.ensure_one()
+        return f"ciusro_{self.name.replace('/', '_')}.xml"
+
+    def _l10n_ro_edi_get_sending_and_failed_documents(self):
+        return self.l10n_ro_edi_document_ids.filtered(lambda d: d.state in ('invoice_sending', 'invoice_sending_failed'))
+
+    def _l10n_ro_edi_compute_errors(self, xml_data):
+        """ Compute possible errors before sending E-Factura """
+        self.ensure_one()
+        errors = []
+        if not self.company_id.l10n_ro_edi_access_token:
+            errors.append(_('Romanian access token not found. Please generate or fill it in the settings.'))
+        if not xml_data:
+            errors.append(_('CIUS-RO XML attachment not found.'))
+        return errors
+
+    def _l10n_ro_edi_send_invoice(self, xml_data):
+        """ Called by _call_web_service_after_invoice_pdf_render in account.move.send.
+            The state flow of Romanian E-Factura invoice goes as follows:
+             - Pre-check any errors from invoice's company before sending, return if found any
+             - Send to E-Factura, and based on the result:
+                 - if error -> delete all other error documents and create a new error document
+                 - if success -> delete all other sending documents and create a new sending document"""
+        self.ensure_one()
+        if errors := self.company_id._l10n_ro_edi_get_errors_pre_request():
+            self._l10n_ro_edi_get_sending_and_failed_documents().unlink()
+            self._l10n_ro_edi_create_document_invoice_sending_failed(self, '\n'.join(errors))
+            return
+
+        result = self.env['l10n_ro_edi.document']._request_ciusro_send_invoice(
+            company=self.company_id,
+            xml_data=xml_data,
+            move_type=self.move_type,
+        )
+        if 'error' in result:
+            self._l10n_ro_edi_get_sending_and_failed_documents().unlink()
+            self._l10n_ro_edi_create_document_invoice_sending_failed(self, result['error'])
+        else:
+            self._l10n_ro_edi_get_sending_and_failed_documents().unlink()
+            self._l10n_ro_edi_create_document_invoice_sending(self, result['key_loading'])
```

Comment:
```suggestion
        self._l10n_ro_edi_get_sending_and_failed_documents().unlink()
        if 'error' in result:
            self._l10n_ro_edi_create_document_invoice_sending_failed(self, result['error'])
        else:
            self._l10n_ro_edi_create_document_invoice_sending(self, result['key_loading'])
```
Do you really want to remove invoice_sending in case of error? I think you only want to remove invoice_sending_failed in that case